### DISCUSSION TITLE
How does C3 copiler work?

### DISCUSSION THREAD

**Discussion by renlite** [c3lang/c3c#1179] [General]
Could you explain the path how the C3 compiler works?
C3 -> C -> native code ?
C3 -> IR -> (LLVM) -> native code?
What about wasm compilation? (Future)

**Comment by lerno** [comment]
C3 -> AST -> LLVM IR -> native code.

C3 can compile to WASM, but the only thing I've verified is compiling for WASM-4

**Comment by renlite** [comment]
Thanks.
Would it be possible to write wasm-modules for this project in C3?
https://github.com/orca-app/orca
As it is written in C, binding should work.

**Comment by lerno** [comment]
It should be possible yes.
